GPyOpt/core/bo.py

```diff
@@ -134,7 +134,8 @@ def run_optimization(self, max_iter = 0, max_time = np.inf,  eps = 1e-8, context
             except np.linalg.linalg.LinAlgError:
                 break
 
-            if not ((self.num_acquisitions < self.max_iter) and (self._distance_last_evaluations() > self.eps)):
```

Comment:
I don't understand why we need to invert all of the comparison operators here. Why not simply add `len(self.X) > 1 and ` bit and leave everything else as is?

Reply:
Because the `not` condition is much more difficult to read. Figured I would fix that in the process. But sure - logically the two are equivalent.